### ISSUE TITLE
Is this library still needed?

### ISSUE THREAD

**Issue by Qqwy** [byteverse/run-st#7]
The Hackage package description reads:

> This package provides specializations of runST that avoid a needless data-constructor allocation for the returned value. If [issue 15127](https://gitlab.haskell.org/ghc/ghc/issues/15127) is resolved, this package will no longer be necessary.

But notably, that GHC issue  has been closed a few years back, specifically by the commit https://gitlab.haskell.org/ghc/ghc/-/commit/46720997a0b1fa2971a884adf43de096ce130a7e

**Comment by andrewthad** [comment]
Yes, this library is no longer needed. Generally, I’ve been a little reluctant to remove it in the places where it had already been used. What this library does cannot possibly make performance worse. From the times that I have removed this from use and before-and-after spot checked the optimized Core that GHC produces, it looks like the optimization that GHC added, does in fact work correctly.

**Comment by 0rphee** [comment]
Should this library be marked as Deprecated and it's README ammended to reflect that?
